src/middlewared/middlewared/plugins/disk.py

```diff
@@ -143,9 +143,7 @@ async def disk_extend(self, disk, context):
             if await self.middleware.call('truenas.is_ix_hardware'):
                 disk['supports_smart'] = True
             else:
-                disk['supports_smart'] = disk['name'].startswith('nvme') or bool(RE_SMART_AVAILABLE.search(
-                    await self.middleware.call('disk.smartctl', disk['name'], ['-a'], {'silent': True}) or ''
-                ))
+                disk['supports_smart'] = disk['name'].startswith('nvme') or await self.middleware.call('disk.smartctl', disk['name'], ['-a', '--json=c'], {'silent': True})['smart_support']['available']
```

Comment:
Looks like this will fail, you're awaiting on `self.middleware.call('disk.smartctl', disk['name'], ['-a', '--json=c'], {'silent': True})['smart_support']['available']`.